### PR TITLE
Make refinement thresholds consistent; expose them through refine API

### LINKED ISSUE #191: GOOD_ENOUGH_THRESH different for numba and not-numba
In feature.py, the GOOD_ENOUGH_THRESH is different for the numba version and the not numba version:

line 199, thresh is set to 0.005 (was previously 0.01 back in February, changed in commit f963d4)
line 303, thresh is set to 0.01

Is this difference intentional?

### LINKED ISSUE #246: Clean up *_THRESH magic numbers.
Make SHIFT_THRESH and GOOD_ENOUGH_THRESH optional arguments of the refine functions and document them. Thanks @sciunto for catching this.

### PR DESCRIPTION
**Pull request by danielballan**
As the title says, this does two separate things:

1. Set  `GOOD_ENOUGH_THRESH` in the numba codepath to the same value it has in the numpy code path. This closes #191 (thanks, @RebeccaWPerry!) This will slightly change the output of the numba code path, but all the tests pass without modification.
2. Stop hard-coding `GOOD_ENOUGH_THRESH` and `SHIFT_THRESH`, instead exposing them as new optional arguments to `refine`. This does not change the output at all. This closes #246 (thanks again, @sciunto )

These parameters are obscure. I welcome opinions on my choice of names ('break_thresh` and `shift_thresh`) and my attempt at explaining them in the docstring.

I intentionally do not expose these parameters through the already-overstuffed `locate`. Users who want to customize these will have to build a processing pipeline the long way.

### CODE REVIEW

**Comment by danielballan** [comment]
P.S. Someday @caspervdw may want to make these new arguments accept tuples. :- )

File: trackpy/feature.py [hunk 1]
```diff
@@ -118,12 +119,21 @@ def refine(raw_image, image, radius, coords, separation=0, max_iterations=10,
         estimated position
     max_iterations : integer
         max number of loops to refine the center of mass, default 10
+    engine : {'python', 'numba'}
```
Comment (tacaswell) [inline trackpy/feature.py]: this should have an `, optional` and include 'auto'

Comment (tacaswell) [inline trackpy/feature.py]: also, `seperation` is not documented

**Comment by caspervdw** [comment]
I went through your changes, :+1: for merging!

File: trackpy/feature.py (hunk 1 above)
Comment (nkeim) [inline trackpy/feature.py]: The `engine='auto'` code is in `locate()`, not here. Suggest we move it here as part of #257.

**Comment by tacaswell** [comment]
I am 50/50 on holding this up over docstrings.  I won't merge it, but won't complain if someone else does.

**Comment by danielballan** [comment]
I'll fix and self-merge right before I tag the RC.

**Pull request merged**